app/workers/detect_invariants.rb

```diff
@@ -23,5 +24,23 @@ def detect_application_choices_in_old_states
     end
   end
 
+  def detect_outstanding_references_on_submitted_applications
+    applications_with_reference_weirdness = ApplicationChoice
+      .joins(application_form: [:application_references])
+      .where.not(application_choices: { status: 'unsubmitted' })
```

Comment:
Are there any other pre-submission statuses we need to consider? I know we've got rid of most of these but does `application_not_sent` count as un-submitted?

In other places we test for the presence of `ApplicationForm#submitted_at`. Would that work?

Reply:
This raises an interesting question actually - should we cancel outstanding references if an application gets [rejected at end of cycle](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/app/services/candidate_interface/cancel_unsubmitted_application_at_end_of_cycle.rb) (and moves to an `application_not_sent` state)? I think we probably should. @tijmenb - WDYT?

A consequence of that is that this query remains unchanged, ie - "only unsubmitted applications should have pending references".

Reply:
I *think* we've cancelled those references (but since deleted the code). We'll see once we run this check!